server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -241,6 +291,26 @@ public SearchPhaseController.ReducedQueryPhase reduce() throws Exception {
 
     }
 
+    private static void consumePartialMergeResult(
```

Comment:
Is this called in the data node, coord node, or both? :)

Reply:
This is only called from `reduce` (i.e. the final reduce) so it's coordinating only.

Reply:
Umm any suggestions? :D I mean it's just a trivial helper to dry up the logic of queuing up the aggs and top docs a bit, always hard to name those well?
I wouldn't worry too much about these parts, I really just went for the shortest change here, we can do this stuff far cleaner in follow-ups IMO, it's kinda buggy to have to copy stuff to a list here to begin with IMO.